Packet++/src/SipLayer.cpp

```diff
@@ -307,8 +307,14 @@ void SipRequestFirstLine::parseVersion()
 	verPos++;
 
 	int endOfVerPos = 0;
-	while (((verPos+endOfVerPos)[0] != '\r') && ((verPos+endOfVerPos)[0] != '\n'))
+	while (((verPos+endOfVerPos)[0] != '\r') && ((verPos+endOfVerPos)[0] != '\n')) {
+		if ((verPos + endOfVerPos) >= (char *) (m_SipRequest->m_Data + m_SipRequest->m_DataLen)) {

```

Comment:
maybe this condition can be added to the while loop?

also, can you please fix the indentation and make sure the opening bracket `{` is in a separate line?

Reply:
Yep, did that.